frontend/src/lib/components/PropertyFilters/components/propertyFilterBetweenLogic.ts

```diff
@@ -0,0 +1,47 @@
+import { actions, kea, key, listeners, path, props, reducers } from 'kea'
+
+import { PropertyFilterValue } from '~/types'
+
+import type { propertyFilterBetweenLogicType } from './propertyFilterBetweenLogicType'
+
+export interface PropertyFilterBetweenLogicProps {
+    value: PropertyFilterValue
+    onSet: (newValue: PropertyFilterValue) => void
+}
+
+function parseNumericValue(value: PropertyFilterValue, index: number): number | null {
+    if (!Array.isArray(value) || value.length !== 2) {
+        return null
+    }
+    const valueAtIndex = Array.isArray(value) ? value[index] : null
+    const numValue = Number(valueAtIndex)
+    return Number.isNaN(numValue) || valueAtIndex === null ? null : numValue
+}
+
+export const propertyFilterBetweenLogic = kea<propertyFilterBetweenLogicType>([
+    props({} as PropertyFilterBetweenLogicProps),
+    key((props) => JSON.stringify(props.value)),
+    path(['lib', 'components', 'PropertyFilters', 'components', 'propertyFilterBetweenLogic']),
+    actions({
+        setLocalMin: (value: number | null) => ({ value }),
+        setLocalMax: (value: number | null) => ({ value }),
+    }),
+    reducers(({ props }) => ({
+        localMin: [
+            parseNumericValue(props.value, 0),
+            {
+                setLocalMin: (_, { value }) => value,
+            },
+        ],
+        localMax: [
+            parseNumericValue(props.value, 1),
+            {
+                setLocalMax: (_, { value }) => value,
+            },
+        ],
+    })),
+    listeners(({ values, props }) => ({
+        setLocalMin: ({ value }) => props.onSet([value ?? NaN, values.localMax ?? NaN]),
+        setLocalMax: ({ value }) => props.onSet([values.localMin ?? NaN, value ?? NaN]),
```

Comment:
Seems odd to me to use `NaN` to represent a missing number. Usually NaN represents the result of invalid math (aka `3/0`), not a missing number. Why not just use `null` or `undefined`?

Reply:
https://github.com/PostHog/posthog/blob/db6fdd1f0496722cf37558d27c8ab678bbf2e57b/frontend/src/types.ts#L801-L802

The existing types don't allow array elements to be `null` or `undefined`. I agree that `NaN` typically isn't used like this, but it felt more safe than changing these types. `NaN` is also the same value set by the LemonInput when a valid number input becomes empty.

Reply:
> `NaN` is also the same value set by the LemonInput when a valid number input becomes empty

👍🏻 Didn't know that. Carry on!